aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -2604,6 +2604,10 @@ Tensor mexp_impl(
 
   if (!compute_highest_degree_approx) {
     auto res = at::empty_like(a, {}, at::MemoryFormat::Contiguous);
+    // To prevent undefined behavior which outputs "normal" result from a matrix
+    // contains NaN values, we put NaN values in `res`, so if input has NaN values,
+    // its computation will be skipped to return the NaN contained `res` directly.
+    res.index_put_({at::indexing::Slice(), 0, 0}, std::numeric_limits<double>::quiet_NaN());
```

Comment:
I don't think I understand this fix. Can you explain it? What happens if the NaN is in the last element of the matrix? How does this fix it?

Reply:
For the batch size = 1 branch, what actually happens is that the operator 1 norm is NaN, and all the `idx_curr_norm_interval` tensors end up empty because NaN is neither above `norm_lower_bound` nor below `norm_lower_bound`. So `res` is returned unchanged. This added line ensures that of the uninitialized memory that gets returned, at least one value will be a NaN. We were thinking initializing the whole `res` to NaN is unnecessarily expensive.

Reply:
What about initialising it as at::full_like(a, NaN, ...)? Doing so will probably be faster on CUDA, as you just launch one kernel. It'd also be a bit less cryptic than this code IMO.

I'm not to fussed about this though, improving the comment LGTM as well.

Reply:
Another option is to do something like:
```
res[torch.isnan(norm)].fill_(torch.nan)
```
At the end of the function. This launches a whole bunch of kernels that all do nothing in the common case though, so maybe `full_like` is a better choice.

Reply:
boolean masking synchronizes, so better to avoid it